src/tree/tree_model.cc

```diff
@@ -617,6 +620,82 @@ std::string RegTree::DumpModel(const FeatureMap& fmap,
   return result;
 }
 
+void RegTree::Load(Json const& in) {
+  fromJson(in["tree_param"], &param);
+  {
+    std::vector<Json> const& v_j_stats = get<Array>(in["stats"]);
+    stats_.resize(v_j_stats.size());
+
+    for (size_t i = 0; i < stats_.size(); ++i) {
+      std::vector<Json> const& j_stat = get<Array const>(v_j_stats[i]);
+      auto& s = stats_[i];
+      s.loss_chg = get<Number const>(j_stat[0]);
+      s.sum_hess = get<Number const>(j_stat[1]);
+      s.base_weight = get<Number const>(j_stat[2]);
+      s.leaf_child_cnt = get<Integer const>(j_stat[3]);
+    }
+    CHECK_EQ(param.num_nodes, stats_.size());
+  }
+
+  {
+    std::vector<Json> const& v_j_nodes = get<Array>(in["nodes"]);
+    nodes_.resize(v_j_nodes.size());
+
+    for (size_t i = 0; i < nodes_.size(); ++i) {
+      auto const& j_node = get<Array const>(v_j_nodes[i]);
+      auto& n = nodes_[i];
+      auto left = get<Integer const>(j_node[0]);
+      auto right = get<Integer const>(j_node[1]);
+      auto parent = get<Integer const>(j_node[2]);
+      auto split_ind = get<Integer const>(j_node[3]);
+      auto split_cond = get<Number const>(j_node[4]);
+      auto default_left = get<Boolean const>(j_node[5]);
+      n = Node(left, right, parent, split_ind, split_cond, default_left);
+    }
+    CHECK_EQ(param.num_nodes, nodes_.size());
+  }
+}
+
+void RegTree::Save(Json* p_out) const {
+  auto& out = *p_out;
+  CHECK_EQ(param.num_nodes, static_cast<int>(nodes_.size()));
+  CHECK_EQ(param.num_nodes, static_cast<int>(stats_.size()));
+  out["tree_param"] = toJson(param);
+  CHECK_EQ(get<String>(out["tree_param"]["num_nodes"]), std::to_string(param.num_nodes));
+
+  std::vector<Json> v_j_stats(stats_.size());
+  using I = Integer::Int;
+
+  for (size_t i = 0; i < stats_.size(); ++i) {
+    auto const& s = stats_[i];
+    std::vector<Json> j_stats(4);
+    j_stats[0] = s.loss_chg;
+    j_stats[1] = s.sum_hess;
+    j_stats[2] = s.base_weight;
+    j_stats[3] = static_cast<I>(s.leaf_child_cnt);
+
+    v_j_stats[i] = std::move(j_stats);
+  }
+  out["stats"] = std::move(v_j_stats);
+
+  std::vector<Json> v_j_nodes(nodes_.size());
+  for (size_t i = 0; i < nodes_.size(); ++i) {
+    auto const& n = nodes_[i];
+    std::vector<Json> j_node(6);
+    j_node[0] = static_cast<I>(n.LeftChild());
+    j_node[1] = static_cast<I>(n.RightChild());
+    j_node[2] = static_cast<I>(n.Parent());
+    j_node[3] = static_cast<I>(n.SplitIndex());
+    j_node[4] = n.SplitCond();
+    j_node[5] = n.DefaultLeft();
+    CHECK(IsA<Boolean>(j_node[5]));
+
+    v_j_nodes[i] = j_node;
```

Comment:
Is here an implicit constructor call to Array? If so could you make it explicit to make it easier to read?